modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node.cc

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/localization/msf/local_map/ndt_map/ndt_map_node.h"
+
+#include "modules/localization/msf/local_map/ndt_map/ndt_map_matrix_handler.h"
+#include "modules/localization/msf/local_map/ndt_map/ndt_map_node_config.h"
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+NdtMapNode::NdtMapNode() {}
+NdtMapNode::~NdtMapNode() {}
+
+void NdtMapNode::Init(const BaseMapConfig* map_config) {
+  map_config_ = map_config;
+
+  map_node_config_ = new NdtMapNodeConfig();
+  map_node_config_->map_version_ = map_config_->GetMapVersion();
+  map_node_config_->has_map_version_ = false;
+  map_node_config_->has_body_md5_ = false;
+  is_reserved_ = false;
+  data_is_ready_ = false;
+  is_changed_ = false;
+  num_valid_cells_ = 0;
+  num_valid_single_cells_ = 0;
+
+  map_matrix_ = new NdtMapMatrix();
```

Comment:
smarter_pointer is preferred. This applies to other pointers too.

Reply:
fixed